elements/lisk-p2p/src/p2p.ts

```diff
@@ -759,26 +764,40 @@ export class P2P extends EventEmitter {
 			random,
 			Math.min(minimumPeerDiscoveryThreshold, knownPeers.length),
 		);
-		/* tslint:enable no-magic-numbers*/
-
-		// TODO: Remove fields that are specific to the current Lisk protocol.
-		const peers = this._pickRandomPeers(randomPeerCount).map(
-			(peerInfo: P2PPeerInfo): ProtocolPeerInfo => {
-				const { ipAddress, ...peerInfoWithoutIp } = peerInfo;
 
-				// The options property is not read by the current legacy protocol but it should be added anyway for future compatibility.
-				return {
-					...peerInfoWithoutIp,
-					ip: ipAddress,
-				};
-			},
-		);
-		const protocolPeerInfoList: ProtocolPeerInfoList = {
-			success: true,
-			peers,
-		};
-
-		request.end(protocolPeerInfoList);
+		if (request.procedure === REMOTE_RPC_GET_BASIC_PEERS_LIST) {
+			const basicPeers = this._pickRandomPeers(randomPeerCount).map(
+				(peerInfo: P2PPeerInfo): P2PPeerInfo =>
+					// Discovery process only require minmal peers data
+					({
+						ipAddress: peerInfo.ipAddress,
+						wsPort: peerInfo.wsPort,
+					}),
+			);
+			const peerInfoList: P2PBasicPeerInfoList = {
+				success: true,
+				peers: basicPeers,
+			};
+			request.end(peerInfoList);
+		} else {
+			// TODO: Remove fields that are specific to the current Lisk protocol.
```

Comment:
I guess this comment doesn't apply anymore, right?

Reply:
@ishantiw could you shed us some light here? I blame this TODO and it comes from https://github.com/LiskHQ/lisk-sdk/commit/7ee21334aad36a7fbd0c244a2042c9785c124c2c

Reply:
I think this comment can be removed